resources/js/ApiPageConnectionRepo.js

```diff
@@ -24,26 +25,36 @@ module.ApiPageConnectionRepo = ( function ( mw, ApiConnectionsBuilder ) {
 
 					this._queryLinks(pagesToAdd).done(
 						function(apiResponse) {
-							this._apiResponseToPagesAndLinks(apiResponse).then(connections => resolve(connections))
+							this._apiResponseToPagesAndLinks(apiResponse, enableDisplayTitle).then(connections => resolve(connections))
 						}.bind(this)
 					);
 				}
 			}.bind(this)
 		);
 	};
 
-	ApiPageConnectionRepo.prototype._apiResponseToPagesAndLinks = function(linkQueryResponse) {
+	ApiPageConnectionRepo.prototype._apiResponseToPagesAndLinks = function(linkQueryResponse, enableDisplayTitle) {
 		return new Promise(
 			function(resolve) {
 				let connections = (new ApiConnectionsBuilder()).connectionsFromApiResponses(linkQueryResponse)
 
-				this._queryPageNodeInfo(connections.pages).done(function(pageInfoResponse) {
+				this._queryPageNodeInfo(connections.pages, enableDisplayTitle).done(function(pageInfoResponse) {
 					let missingPages = Object.values(pageInfoResponse.query.pages)
 						.filter(p => p.missing === '')
 						.map(p => p.title);
 
+					let displaytitles = [];
+					pageInfoResponse.query.pages.forEach(function(page) {
+						if ( page.pageprops && page.pageprops.displaytitle) {
+							displaytitles[page.title] = page.pageprops.displaytitle;
+						} else {
+							displaytitles[page.title] = page.title;
+						}
+					});
+
 					connections.pages.forEach(function(page) {
-						if(missingPages.includes(page.title)) {
+						page.displaytitle = displaytitles[page.title];
```

Comment:
why not displayTitle instead of displaytitle everywhere?

Reply:
Done.